bin/test-cleanup

```diff
@@ -11,11 +11,15 @@ Usage: ${FUNCNAME[0]} TARGET_ARRAY RESOURCE_TYPE LINKERD_LABEL (FRIENDLY_NAME)"
     return
   }
 
-  local -n res=$1
-  while IFS= read -r line; do res+=("$line"); done <<< "$(kubectl --context="$k8s_context" get "$2" -oname -l "linkerd.io/$3")"
-  if [ -z "${res[*]}" ]; then
-    echo "no ${4:-$2} found" >&2
-  fi
+  # We are using the variable i even if shellsheck does not see it
+  # shellcheck disable=SC2034
+  local i=0
+  unset -v "$1"
+  while IFS= read -r "$1[i++]"; do :; done <<< "$(kubectl --context="$k8s_context" get "$2" -oname -l "linkerd.io/$3")"
```

Comment:
I noticed while testing this that a Linkerd rolebinding ends up in `kube-system` and is not removed:

```
❯ kubectl get -A rolebindings 
NAMESPACE     NAME                                                ROLE                                                  AGE
..
kube-system   linkerd-linkerd-tap-auth-reader                     Role/extension-apiserver-authentication-reader        103s
..
```

This is an existing issue so can we just fix it here by checking all namespaces?

```suggestion
  while IFS= read -r "$1[i++]"; do :; done <<< "$(kubectl --context="$k8s_context" get -A "$2" -oname -l "linkerd.io/$3")"
```